xray/commands/audit/jas/common.go

```diff
@@ -88,46 +99,60 @@ func deleteJasProcessFiles(configFile string, resultFile string) error {
 	}
-	return results, nil
+	return
 }
 
-func convertSarifResultsToSourceCodeScanResults(sarifResults []*sarif.Result, workingDir string, scanType utils.JasScanType) []*utils.SourceCodeScanResult {
-	var sourceCodeScanResults []*utils.SourceCodeScanResult
-	for _, sarifResult := range sarifResults {
-		// Describes a request to “suppress” a result (to exclude it from result lists)
+func processJasScanRun(sarifRun *sarif.Run, workingDir string) *sarif.Run {
+	processed := sarif.NewRun(sarifRun.Tool)
+	// Jas reports has only one invocation
+	invocation := sarifRun.Invocations[0]
+	// Set the actual working directory to the invocation, not the analyzerManager directory
+	// Also used to calculate relative paths if needed with it
+	invocation.WorkingDirectory.WithUri(workingDir)
+	processed.Invocations = append(processed.Invocations, invocation)
+	// Process results
+	for _, sarifResult := range sarifRun.Results {
 		if len(sarifResult.Suppressions) > 0 {
+			// Describes a request to “suppress” a result (to exclude it from result lists)
 			continue
 		}
-		// Convert
-		currentResult := utils.GetResultIfExists(sarifResult, workingDir, sourceCodeScanResults)
-		if currentResult == nil {
-			currentResult = utils.ConvertSarifResultToSourceCodeScanResult(sarifResult, workingDir)
-			// Set specific Jas scan attributes
-			if scanType == utils.Secrets {
-				currentResult.Text = hideSecret(utils.GetResultLocationSnippet(sarifResult.Locations[0]))
+		processed.Results = append(processed.Results, sarifResult)
+		if rule, err := sarifRun.GetRuleById(*sarifResult.RuleID); err == nil {
+			// Add to the rule security-severity score base on results severity
+			score := convertToScore(utils.GetResultSeverity(sarifResult))
+			if score != utils.MissingCveScore && rule.Properties == nil {
+				properties := sarif.NewPropertyBag()
+				properties.Add("security-severity", score)
+				rule.WithProperties(properties.Properties)
 			}
-			sourceCodeScanResults = append(sourceCodeScanResults, currentResult)
-		}
-		if scanType == utils.Sast {
-			currentResult.CodeFlow = append(currentResult.CodeFlow, utils.GetResultCodeFlows(sarifResult, workingDir)...)
 		}
 	}
-	return sourceCodeScanResults
+	return processed
+}
+
+func convertToScore(severity string) string {
+	if level, ok := mapSeverityToScore[strings.ToLower(severity)]; ok {
+		return level
+	}
+	return ""
+}
+
+func GetJasMarkdownDescription(scanType utils.JasScanType, location *sarif.Location, severity, content string) string {
+	dataColumnHeader := "Finding"
+	if scanType == utils.Secrets {
+		dataColumnHeader = "Secret"
+	}
+	headerRow := fmt.Sprintf("| Severity | File | Line:Column | %s |\n", dataColumnHeader)
+	separatorRow := "| :---: | :---: | :---: | :---: |\n"
+	tableHeader := headerRow + separatorRow
+	return tableHeader + fmt.Sprintf("| %s | %s | %s | %s |", severity, utils.GetLocationFileName(location), utils.GetStartLocationInFile(location), content)
```

Comment:
this is frogbot only code, Can we just move it to frogbot?

Reply:
done